recipe/features.py

```diff
@@ -52,19 +52,32 @@ def search_recipes(query):
 
 
 #is used to create or add to the user history
+from django.core.exceptions import MultipleObjectsReturned
+
 def create_or_add_to_history(request, recipe):
     if request.user.is_authenticated:
-        user_history,created = UserHistory.objects.get_or_create(user=request.user)
-        if not user_history.recipe.filter(pk=recipe.pk).exists():
-            user_history.recipe.add(recipe)
+        try:
+            user_history = UserHistory.objects.get(user=request.user)
+        except UserHistory.DoesNotExist:
+            user_history = UserHistory.objects.create(user=request.user)
+        except MultipleObjectsReturned:
```

Comment:
There should not be multiple object, as once history is created, object are simply added. Instead of handle this exception, it must be eliminated.

Reply:
i am getting error eveytime MutipleObjectReturned

Reply:
Let me check!

Reply:
Unable to reproduce in local setup.